driver-core/src/main/java/com/datastax/driver/core/Metadata.java

```diff
@@ -160,7 +184,7 @@ static String fullFunctionName(String simpleName, Collection<?> argumentTypes) {
      * or even {@link Cluster#connect(String)}.
      */
     public static String quote(String id) {
-        return '"' + id + '"';
+        return '"' + id.replace("\"", "\"\"") + '"';
```

Comment:
This is unrelated, but I think it's a bug we never came across so far. An identifier such as `foo " bar` would not be correctly transformed into `foo "" bar`. I modified a few tests in UserTypesTest to show evidence of it.